app/src/main/java/com/duckduckgo/app/bookmarks/ui/FavoritesAdapter.kt

```diff
@@ -171,9 +221,10 @@ sealed class FavoritesScreenViewHolders(itemView: View) : RecyclerView.ViewHolde
             }
         }
 
-        private fun loadFavicon(url: String) {
+        private fun loadFavicon(url: String, image: ImageView) {
             lifecycleOwner.lifecycleScope.launch {
-                faviconManager.loadToViewFromLocalOrFallback(url = url, view = itemView.findViewById(CommonR.id.image))
+                favIconRequestsChannel.send(FavIconRequest(url))
```

Comment:
Not 100% sure about this. Each time we send something to the channel it will access the disk even when is not needed. I think `faviconManager.loadToViewFromLocalOrFallback` could return true/false if a favicon already exists and in that case don't send to the channel.

Reply:
Thanks! Updated to not send the event when binding the view :+1: